src/serializers/type_serializers/function.rs

```diff
@@ -299,6 +299,7 @@ impl BuildSerializer for FunctionWrapSerializer {
             let schema_copy = schema.copy()?;
             // remove the serialization key from the schema so we don't recurse
             schema_copy.del_item(intern!(py, "serialization"))?;
+            schema_copy.del_item(intern!(py, "ref")).ok(); // remove ref if it exists
```

Comment:
I don't know if this is sensible. It does fix the issue, and it sort of makes sense to me that copies of the schema shouldn't have refs, but beyond that, I don't have a great reason for why this is the right thing to do.